tempesta_fw/http2/hpack_decoder.c

```diff
@@ -0,0 +1,648 @@
+					printf("\"\n");
+#endif
+					if (unlikely(n == 0)) {
+						state &= ~HPack_State_Mask;
+						state |= HPack_State_Value;
+					}
+				}
+			case HPack_State_Value:
+				{
+					uchar c;
+
+ Get_Value:
+					DPUTS("Decode header value length...");
+					if (unlikely(m == 0)) {
+						src = buffer_next(source, &m);
+					}
+					c = *src++;
+					n--;
+					m--;
+					length = c & 0x7F;
+					if (c & 0x80) {
+						DPRINTF("Name length: %" PRIuPTR
+							"\n", length);
+						state |=
+						    HPack_Flags_Huffman_Value;
+					}
+					if (unlikely(length == 0x7F)) {
+						state &= ~HPack_State_Mask;
+						state |=
+						    HPack_State_Value_Length;
+						GET_FLEXIBLE(length);
+					}
+					DPRINTF("Value length: %" PRIuPTR "\n",
+						length);
+					field->value.len = length;
+					if (n >= length) {
+						goto Get_Value_Text;
+					} else {
+						state &= ~HPack_State_Mask;
+						state |= HPack_State_Value_Text;
+						goto Incomplete;
+					}
+				}
+			case HPack_State_Value_Text:
+				{
+					uchar *dst;
+					unsigned int hrc;
+
+					length = field->value.len;
+					if (unlikely(n < length)) {
+						DPUTS("Not enough data...");
+						break;
+					}
+ Get_Value_Text:
+					DPUTS("Decode header value...");
+					if (length) {
+						if ((state & HPack_Flags_Pseudo)
+						    == 0) {
+							/* Add ":" between name and value: */
+							dst =
+							    buffer_small(buffer,
+									 2, 1);
+							dst[0] = ':';
+							dst[1] = ' ';
+						}
+						if (state &
+						    HPack_Flags_Huffman_Value) {
+							DPRINTF("Decode %"
+								PRIuPTR
+								" bytes of Huffman data...\n",
+								length);
+							buffer_close(source, m);
+							hrc =
+							    huffman_decode_fragments
+							    (source, buffer,
+							     length);
+							if (unlikely(hrc)) {
+								*rc = hrc;
+								goto Bug;
+							}
+							field->value =
+							    buffer->str;
+							n -= length;
+							if (likely(n)) {
+								src =
+								    buffer_get
+								    (source,
+								     &m);
+							}
+						} else {
+							DPRINTF("Copy %" PRIuPTR
+								" bytes of plain text...\n",
+								length);
+							HPACK_COPY_STRING
+							    (value);
+						}
+#if Debug_HPack
+						printf("Decoded value: \"");
+						buffer_str_print(&field->value);
+						printf("\"\n");
+#endif
+					}
+#if Debug_HPack
+					else {
+						DPUTS("Zero-length value");
+					}
+#endif
+					if (index) {
+ Duplicate:
+						DPRINTF
+						    ("Add header by dictionary index: %"
+						     PRIuPTR "\n", index);
+#if Fast_Capacity > 32
+						if ((index >> 32) == 0) {
+							hrc =
+							    hpack_add_index(hp->
+									    dynamic,
+									    field,
+									    index,
+									    state,
+									    buffer);
+						} else {
+							hrc =
+							    Err_HTTP2_IntegerOveflow;
+						}
+#else
+						hrc =
+						    hpack_add_index(hp->dynamic,
+								    field,
+								    index,
+								    state,
+								    buffer);
+#endif
```

Comment:
I cannot find the static or dynamic table searching during 'hpack_decode()' procedure - only adding indexes into dynamic table.